File: 프로그래머스/1/42748. K번째수/K번째수.java
import java.util.Arrays;

class Solution {
    public int[] solution(int[] array, int[][] commands) {
        int[] answer = new int[commands.length];
        int index = 0; 
        
        for (int[] command : commands) {
            int i = command[0];
            int j = command[1];
            int k = command[2];
            
            int[] sliced = Arrays.copyOfRange(array, i - 1, j);
            
            Arrays.sort(sliced);
            
            answer[index] = sliced[k - 1];
            index++;
        }
        return answer;
    }
}

File: 프로그래머스/1/42748. K번째수/K번째수.cpp
#include <vector>
#include <algorithm>

using namespace std;

vector<int> solution(vector<int> array, vector<vector<int>> commands) {
    vector<int> answer;
    
    for (int c = 0; c < commands.size(); c++) {
        int i = commands[c][0];
        int j = commands[c][1]; 
        int k = commands[c][2];
        
        vector<int> temp(array.begin() + (i - 1), array.begin() + j);
        
        sort(temp.begin(), temp.end());
        
        answer.push_back(temp[k - 1]);
    }
    
    return answer;
}